### ISSUE TITLE
nvcc.exe console popping up while running cupy code which have been cythonized

### ISSUE THREAD

**Issue by monzelr** [cupy/cupy#8431]
### Description

I have a python code that uses cupy. This code is being cythonized and built during deployment. After that while running the deployed code whenever cupy needs to compile .cu files (and make use of nvvc.exe), nvvc.exe keep opening and closing in a console on it's own. This is an undesired behavior for a desktop application with an user interface.

### To Reproduce

It is difficult to reproduce since I do not know which cupy functions needs just-in-time compilation of cu files. If you know a function, I can make a small python package, which will be cythonized so we have an example.

### Installation

Wheel (`pip install cupy-***`)

### Environment

```
OS                           : Windows-10-10.0.19045-SP0
Python Version               : 3.8.10
CuPy Version                 : 12.3.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.24.4
SciPy Version                : 1.10.1
Cython Build Version         : 0.29.36
Cython Runtime Version       : 0.29.23
CUDA Root                    : C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6
nvcc PATH                    : C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\bin\nvcc.EXE
CUDA Build Version           : 11080
CUDA Driver Version          : 11080
CUDA Runtime Version         : 11060
cuBLAS Version               : (available)
cuFFT Version                : 10600
cuRAND Version               : 10209
cuSOLVER Version             : (11, 3, 2)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 6)
Thrust Version               : 101501
CUB Build Version            : 101501
Jitify Build Version         : 4a37de0
cuDNN Build Version          : (not loaded; try `import cupy.cuda.cudnn` first)
cuDNN Version                : (not loaded; try `import cupy.cuda.cudnn` first)
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA T500
Device 0 Compute Capability  : 75
Device 0 PCI Bus ID          : 0000:01:00.0

```

### Additional Information

To solve this problem I have added the argument `creationflags=subprocess.CREATE_NO_WINDOW` in the `subprocess.check_output` function in "\cupy\cuda\compiler.py" (https://github.com/cupy/cupy/blob/dd37c98701e57d305ab6f1794201dceb91010f53/cupy/cuda/compiler.py#L62).

Since then, the console nvcc.exe does not pop up anymore. Can someone verify this?

**Comment by monzelr** [comment]
Note here: for cupy versions < 9.6.0 and cuda toolkit < 11.0 I did not face the problem with the nvcc.exe console...

**Comment by leofang** [comment]
If you want to avoid to the maximal extent the invocation of nvcc on Windows, I suggest you to update your Python to at least 3.9 (you're using 3.8) and CuPy to at least v13.0.0 (you're using v12.3.0). This issue is a duplicate of #8380 and has been resolved.

However, normally nvcc should not open a new console.. something is still odd on your system 😅

**Comment by monzelr** [comment]
Ok - I will try python 3.9 and cupy >= v13 and give a report of this issue. Thank you for the feedback.

**Comment by monzelr** [comment]
I have installed python 3.9 with cupy v13.2.0. The cythonized code works without the nvcc.exe console popping up.
v13 works like a charm and we see performance improvements - a really great job by the cupy developers. big thanks!

This nvcc.exe issue exists for cupy versions v10, v11, v12 and it was the reason for us to stay at v9.

I guess I can close the PR #8432 because it is resolved since v13?

**Comment by leofang** [comment]
Thanks for giving it a try and sharing your feedback, @monzelr! Glad to know it works better now.
> I guess I can close the PR #8432 because it is resolved since v13?

I'd suggest to wait for the CuPy team for final decision. Maybe @kmaehashi can make a call. Strictly speaking It's not really "resolved" in v13, just that the chance of having to call nvcc is significantly reduced for many use cases, but if nvcc is called (saying using [`RawKernel`](https://docs.cupy.dev/en/stable/user_guide/kernel.html#raw-kernels) or [`RawModule`](https://docs.cupy.dev/en/stable/user_guide/kernel.html#raw-modules) with `backend="nvcc"`) you should still see a console popping out without your change. Feel free to give it a shot and confirm it's the case 🙂

**Comment by kmaehashi** [comment]
It's nice to have! Let's proceed to merge 😃

**Comment by leofang** [comment]
Closed by #8432.
